server.go

```diff
@@ -2797,15 +2798,62 @@ func (s *server) peerConnected(conn net.Conn, connReq *connmgr.ConnReq,
 	// offered that would trigger channel closure. In case of outgoing
 	// htlcs, an extra block is added to prevent the channel from being
 	// closed when the htlc is outstanding and a new block comes in.
-	p, err := newPeer(
-		s.cfg, conn, connReq, s, peerAddr, inbound, initFeatures,
-		legacyFeatures, s.cfg.ChanEnableTimeout,
-		lncfg.DefaultOutgoingCltvRejectDelta, errBuffer,
-	)
-	if err != nil {
-		srvrLog.Errorf("unable to create peer %v", err)
-		return
-	}
+	pCfg := ppeer.Config{
```

Comment:
it's beautiful 😥

nit: `peerConfig`?

Reply:
Will have to move some lines around if we do this (the copying of public keys right before the call to `newPeer` / `NewBrontide` depending on the commit. Is it worth it?

Reply:
Eh I don't mind the way it is, plus would have to move some lines around bc we have to copy the keys and they'd be too long.

Reply:
sgtm